### PR TITLE
Update cellPartOfAnatomicalEntity.yaml

### PR DESCRIPTION
**Pull request by dosumis**
Add back xrefs.

### CODE REVIEW

**Comment by bvarner-ebi** [comment]
@matentzn, @dosumis, this PR has been approved, but not yet merged. Last action Oct 2021.
Can we merge or close?
Will also add to the "DOS" board.

**Comment by matentzn** [comment]
@bvarner-ebi if you merge master in to this branch and run:

```
sh run.sh make ../patterns/definitions.owl -B
```

We will know for sure if it works.

**Comment by bvarner-ebi** [comment]
```
% sh run.sh make ../patterns/definitions.owl -B
if [ true  = true ] && [ "cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" ]; then dosdp-tools generate --catalog=catalog-v001.xml \
    --infile=../patterns/data/default/ --template=../patterns/dosdp-patterns --batch-patterns="cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" \
    --ontology=cl-edit.owl --obo-prefixes=true --outfile=../patterns/data/default; fi
2022-08-25 08:19:03.239Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <cell>
2022-08-25 08:19:03.253Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <biological_process>
2022-08-25 08:19:03.700Z error [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] JSON does not conform to DOS-DP schema for pattern file at ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml:
String: DownField(xrefs),DownField(def)
2022-08-25 08:19:03.710Z error [] [context: ] JSON does not conform to DOS-DP schema for pattern file at ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml
make: *** [Makefile:544: ../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Error 1
```

**Comment by matentzn** [comment]
Yeah, @rays22 can help you with that

**Comment by rays22** [comment]
> 2022-08-25 08:19:03.710Z error [] [context: ] JSON does not conform to DOS-DP schema for pattern file at ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml
> make: *** [Makefile:544: ../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Error 1

I think I have fixed it, but running
```
sh run.sh make ../patterns/definitions.owl -B
```
seems to take ages.

**Comment by bvarner-ebi** [comment]
> I think I have fixed it, but running
> 
> ```
> sh run.sh make ../patterns/definitions.owl -B
> ```
> 
> seems to take ages.

Thank you, @rays22. I ran it, and it took a couple of minutes.

I still get missing column messages:

```
ontology % sh run.sh make ../patterns/definitions.owl -B
if [ true  = true ] && [ "cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" ]; then dosdp-tools generate --catalog=catalog-v001.xml \
    --infile=../patterns/data/default/ --template=../patterns/dosdp-patterns --batch-patterns="cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" \
    --ontology=cl-edit.owl --obo-prefixes=true --outfile=../patterns/data/default; fi
2022-08-26 11:31:57.101Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <cell>
2022-08-26 11:31:57.115Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <biological_process>
2022-08-26 11:31:57.566Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <cell>
2022-08-26 11:31:57.566Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <anatomical_entity>
2022-08-26 11:31:57.566Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <pubs>
if [ true  = true ] && [ "cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" ]; then dosdp-tools generate --catalog=catalog-v001.xml \
    --infile=../patterns/data/default/ --template=../patterns/dosdp-patterns --batch-patterns="cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" \
    --ontology=cl-edit.owl --obo-prefixes=true --outfile=../patterns/data/default; fi
2022-08-26 11:32:23.442Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <cell>
2022-08-26 11:32:23.455Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml;input=../patterns/data/default//cellCapableOfBiologicalProcess.tsv;output=../patterns/data/default/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/default//cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns/cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <biological_process>
2022-08-26 11:32:23.886Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <cell>
2022-08-26 11:32:23.887Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <anatomical_entity>
2022-08-26 11:32:23.887Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml;input=../patterns/data/default//cellPartOfAnatomicalEntity.tsv;output=../patterns/data/default/cellPartOfAnatomicalEntity.ofn] Input ../patterns/data/default//cellPartOfAnatomicalEntity.tsv for pattern ../patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml is missing column for pattern variable <pubs>
if [ true  = true ] && [ "cellCapableOfBiologicalProcess" ]; then dosdp-tools generate --catalog=catalog-v001.xml \
    --infile=../patterns/data/clustering --template=../patterns/dosdp-patterns/ --batch-patterns="cellCapableOfBiologicalProcess" \
    --ontology=cl-edit.owl --obo-prefixes=true --outfile=../patterns/data/clustering; fi
2022-08-26 11:32:49.361Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns//cellCapableOfBiologicalProcess.yaml;input=../patterns/data/clustering/cellCapableOfBiologicalProcess.tsv;output=../patterns/data/clustering/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/clustering/cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns//cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <cell>
2022-08-26 11:32:49.378Z  warn [] [context: command=generate;pattern=../patterns/dosdp-patterns//cellCapableOfBiologicalProcess.yaml;input=../patterns/data/clustering/cellCapableOfBiologicalProcess.tsv;output=../patterns/data/clustering/cellCapableOfBiologicalProcess.ofn] Input ../patterns/data/clustering/cellCapableOfBiologicalProcess.tsv for pattern ../patterns/dosdp-patterns//cellCapableOfBiologicalProcess.yaml is missing column for pattern variable <biological_process>
if [ true  = true ] && [ "cellCapableOfBiologicalProcess cellPartOfAnatomicalEntity" ]   || [ "cellCapableOfBiologicalProcess" ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -i ../patterns/data/default/cellCapableOfBiologicalProcess.ofn -i ../patterns/data/default/cellPartOfAnatomicalEntity.ofn -i ../patterns/data/clustering/cellCapableOfBiologicalProcess.ofn \
	annotate --ontology-iri http://purl.obolibrary.org/obo/cl/patterns/definitions.owl  --version-iri http://purl.obolibrary.org/obo/cl/releases/2022-08-26/patterns/definitions.owl \
      --annotation owl:versionInfo 2022-08-26 -o definitions.ofn && mv definitions.ofn ../patterns/definitions.owl; fi
```

**Comment by matentzn** [comment]
You could make a dosdp tools issue https://github.com/INCATools/dosdp-tools/issues

Linking to this PR, saying that

## Redundant missing column message if table is empty

It seems like if the dosdp table has no data, dosdp-tools erroneously believes that it is missing a column

File: src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml [hunk 1]
```diff
@@ -2,25 +2,30 @@ pattern_name: cellPartOfAnatomicalEntity
 pattern_iri: http://purl.obolibrary.org/obo/cl/cellPartOfAnatomicalEntity
 
 description: A cell that is part of an anatomical entity.
-#examples: CL_0009001 compound eye retinal cell, CL_2000004 pituitary gland cell (654 total)
+# examples:
```
Comment (dosumis) [inline src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml:5]: This should guide usage of the template.  Please change to:

"Use this template to define a class of cells by a combination of gross type and location.  e.g. Peyer's patch B cell.|

File: src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml [hunk 2]
```diff
@@ -2,25 +2,30 @@ pattern_name: cellPartOfAnatomicalEntity
...
+# CL_0009001 compound eye retinal cell,
+# CL_2000004 pituitary gland cell (654 total)
```
Comment (dosumis) [inline src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml:7]: We should be able to manage to some better examples.  I don't like these 'any cell type' X location terms.

"Peyer's patch B cell"   is better than the pituitary gland example.  The compound eye example is really just a way of burying explicit taxon declaration, which also makes it a slightly odd example.

File: src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml [hunk 3]
```diff
@@ -34,8 +39,9 @@ def:
   vars:
     - cell
     - anatomical_entity
-#  xrefs:
-#    value: pubs
+  annotations:
+    - annotationProperty: xref
+      value: pubs
```
Comment (dosumis) [inline src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml:44]: This  works, but can all be simplified by using the built-in xref key. (Earlier attempt at this seems to have been deleted).  Happy to revisit this in a second PR though.

**Comment by shawntanzk** [comment]
@bvarner-ebi - assigning this to you, and @rays22 can help if you need any. Thanks

**Comment by bvarner-ebi** [comment]
> @bvarner-ebi - assigning this to you, and @rays22 can help if you need any. Thanks

Not sure what needs to be done here... I tried to close this PR, but unclear to me what the background is as there is no linked issue. I opened [a ticket](https://github.com/INCATools/dosdp-tools/issues/429) per @matentzn's direction, but next steps are not clear to me.

As this has been a pending PR since 2021 with sparse follow up, I'm inclined to close this PR without merging.

**Comment by matentzn** [comment]
@bvarner-ebi can you confirm that @dosumis comments where addressed, if so, just merge it! Its a totally fine PR!

**Comment by rays22** [comment]
> Not sure what needs to be done here... I tried to close this PR, but unclear to me what the background is as there is no linked issue. I opened [a ticket](https://github.com/INCATools/dosdp-tools/issues/429) per @matentzn's direction, but next steps are not clear to me.
> 
> As this has been a pending PR since 2021 with sparse follow up, I'm inclined to close this PR without merging.

@bvarner-ebi , I think I know what is left to be done for this PR. I will update the `src/patterns/dosdp-patterns/cellPartOfAnatomicalEntity.yaml` DOS-DP template and take care of merging the PR.

**Review by matentzn — APPROVED** [review]

**Pull request merged**